File: loja-java/src/test/java/loja/TestLoja.java
package loja;

import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.assertEquals;

public class TestLoja {

    @Test
    void carrinhoDeveSomarProdutos() {

        Carrinho carrinho = new Carrinho();

        carrinho.adicionar(new Produto("Livro", 50.0));
        carrinho.adicionar(new Produto("Caneta", 5.0));

        assertEquals(2, carrinho.quantidade());
        assertEquals(55.0, carrinho.total());
    }

    @Test
    void produtoDeveConterNomeEPreco() {

        Produto produto = new Produto("Livro", 50.0);

        assertEquals("Livro", produto.nome());
        assertEquals(50.0, produto.preco());
    }
}

File: loja-ruby/test/test_loja.rb
require "minitest/autorun"
require_relative "../lib/loja/produto"
require_relative "../lib/loja/carrinho"

class LojaTest < Minitest::Test
  def test_total_do_carrinho
    carrinho = Loja::Carrinho.new
    carrinho.adicionar(Loja::Produto.new("Livro", 50.0))
    carrinho.adicionar(Loja::Produto.new("Caneta", 5.0))

    assert_equal 2, carrinho.quantidade
    assert_equal 55.0, carrinho.total
  end
end
